File: unix/boot/spp/rpp/ratlibf/dsinit.f
      subroutine dsinit (w)
      integer w
      integer mem( 60000)
      common/cdsmem/mem
      integer t
      if (.not.(w .lt. 2 * 2 + 2))goto 23000
      call error (42Hin dsinit: unreasonably small memory size.)
23000 continue
      t = 2
      mem (t + 0) = 0
      mem (t + 1) = 2 + 2
      t = 2 + 2
      mem (t + 0) = w - 2 - 1
      mem (t + 1) = 0
      mem (1) = w
      return
      end

File: unix/boot/spp/rpp/ratlibr/dsinit.r
include	defs

# dsinit --- initialize dynamic storage space to w words

   subroutine dsinit (w)
   integer w

   DS_DECL(Mem, MEMSIZE)

   pointer t

   if (w < 2 * DS_OHEAD + 2)
      call error ("in dsinit: unreasonably small memory size.")

   # set up avail list:
   t = DS_AVAIL
   Mem (t + DS_SIZE) = 0
   Mem (t + DS_LINK) = DS_AVAIL + DS_OHEAD

   # set up first block of space:
   t = DS_AVAIL + DS_OHEAD
   Mem (t + DS_SIZE) = w - DS_OHEAD - 1     # -1 for MEMEND
   Mem (t + DS_LINK) = LAMBDA

   # record end of memory:
   Mem (DS_MEMEND) = w

   return
   end
